### DISCUSSION TITLE
Folder structure

### DISCUSSION THREAD

**Discussion by amos-kibet** [3Stones-io/omedis#398] [General]
A resource can have one or more checks/change/validate/ etc modules. Currently we don't seem to have a convention on where to go about this.

![Screenshot 2024-11-27 at 10 47 30](https://github.com/user-attachments/assets/4c877852-5443-42b7-b265-c1e39195eaaa)

Do we group these modules per resource, or globally, i.e. all change modules, etc,  are namespaced under a parent `change` module?

cc @almirsarajcic, @kagure-nyakio, @wintermeyer

**Comment by kagure-nyakio** [comment]
Not sure, but maybe this might help: https://hexdocs.pm/ash/project-structure.html

**Comment by almirsarajcic** [comment]
Good catch, Amos!
Yeah, I agree these should be under the resource's namespace. Thanks, Nyakio, for the link.
If Stefan also agrees, then create a task for moving these around.

**Comment by amos-kibet** [comment]
@wintermeyer, what's your take on this?

**Reply by kagure-nyakio** [reply]
I think this is a part of what I am working on in #418

**Reply by amos-kibet** [reply]
Agreed!

**Comment by wintermeyer** [comment]
I don't care how this is done as long as it is done in the spirit of the framework. I don't have to agree with the framework. But when I use it I opt into its soul otherwise I shouldn't use it all together. Long story short: https://hexdocs.pm/ash/project-structure.html is the way to go.

@amos-kibet Thank you for bringing this up! These details make a great software project.
